src/components/templates/discount-table/use-promotion-columns.tsx

```diff
@@ -0,0 +1,223 @@
+  const columns = useMemo(
+    () => [
+      {
+        Header: <Table.HeadCell className="pl-2">Code</Table.HeadCell>,
+        accessor: "code",
+        Cell: ({ cell: { value }, index }) => (
+          <Table.Cell key={index}>
+            <div className="overflow-hidden">
+              <Badge className="rounded-rounded" variant="default">
+                <span className="inter-small-regular">{value}</span>
+              </Badge>
+            </div>
+          </Table.Cell>
+        ),
+      },
+      {
+        Header: "Description",
+        accessor: "rule.description",
+        Cell: ({ cell: { value }, index }) => (
+          <Table.Cell key={index}>{value}</Table.Cell>
+        ),
+      },
+      {
+        Header: <div className="text-right">Amount</div>,
+        id: "amount",
+        Cell: ({ row: { original }, index }) => {
+          return (
+            <Table.Cell className="text-right" key={index}>
+              {getPromotionAmount(original)}
+            </Table.Cell>
+          )
+        },
+      },
+      {
+        Header: <div className="w-[60px]" />,
+        id: "currency",
+        Cell: ({ row: { original }, index }) => (
+          <Table.Cell className="px-2 text-grey-40" key={index}>
+            {getCurrencySymbol(original)}
+          </Table.Cell>
+        ),
+      },
+      {
+        Header: "Status",
+        accessor: "ends_at",
+        Cell: ({ row: { original }, index }) => (
+          <Table.Cell key={index}>{getPromotionStatusDot(original)}</Table.Cell>
+        ),
+      },
+      {
+        Header: () => <div>Products</div>,
+        id: "products",
+        Cell: ({ row: { original }, index }) => {
+          let count = 0
+          let text
+          if (!original.rule.conditions?.length) {
+            text = "All products"
+          } else {
+            text = getConditionText(original.rule.conditions[0])
+            count = original.rule.conditions.reduce((prev, curr) => {
+              return prev + getNumberOfConditions(curr)
+            }, -1)
+          }
+
+          return (
+            <Table.Cell className="" key={index}>
+              {text}{" "}
+              {count > 0 ? (
+                <span className="text-grey-40">{` + ${count} more`}</span>
+              ) : (
+                <></>
+              )}
+            </Table.Cell>
+          )
+        },
+      },
+      {
+        Header: () => <div className="text-right">Redemptions</div>,
+        accessor: "usage_count",
+        Cell: ({ row: { original }, index }) => {
+          return (
+            <Table.Cell className="text-right" key={index}>
+              {original.usage_limit > 0
+                ? getUsageCount(original.usage_count)
+                : "-"}
+            </Table.Cell>
+          )
+        },
+      },
+    ],
+    []
+  )
+
+  return [columns]
+}
+
+const getUsageCount = (usageCount: number) => {
+  switch (true) {
+    case usageCount > 9999999:
```

Comment:
A different solution could be something like this: 

I prefer the switch because if the immediate readability

```
  const factor = Math.floor((`${usageCount}`.length - 1) / 3)
  const units = ["", "k", "m", "b"]
  return `${Math.floor(usageCount / Math.pow(10, 3 * factor))}${units[factor]}`
```

Reply:
yeah, I agree, highly unlikely, though I thought it better to just cover our bases 😄